Description: Use the shortname for GCE
 GCE FDQN's may exceed 64 characters. A number of programs like Hadoop,
 Easyrsa, Java, etc., have issues with long hostnames.
Author: Ben Howard <ben.howard@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1383794
--- a/cloudinit/sources/DataSourceGCE.py
+++ b/cloudinit/sources/DataSourceGCE.py
@@ -115,7 +115,8 @@
         return self.metadata['public-keys']
 
     def get_hostname(self, fqdn=False, _resolve_ip=False):
-        return self.metadata['local-hostname']
+        # GCE has long FDQN's and has asked for short hostnames
+        return self.metadata['local-hostname'].split('.')[0]
 
     def get_userdata_raw(self):
         return self.metadata['user-data']
--- a/tests/unittests/test_datasource/test_gce.py
+++ b/tests/unittests/test_datasource/test_gce.py
@@ -80,7 +80,8 @@
             body=_request_callback)
         self.ds.get_data()
 
-        self.assertEqual(GCE_META.get('instance/hostname'),
+        shostname = GCE_META.get('instance/hostname').split('.')[0]
+        self.assertEqual(shostname,
                          self.ds.get_hostname())
 
         self.assertEqual(GCE_META.get('instance/id'),
@@ -107,5 +108,5 @@
         self.assertEqual(GCE_META_PARTIAL.get('instance/id'),
                          self.ds.get_instance_id())
 
-        self.assertEqual(GCE_META_PARTIAL.get('instance/hostname'),
-                         self.ds.get_hostname())
+        shostname = GCE_META_PARTIAL.get('instance/hostname').split('.')[0]
+        self.assertEqual(shostname, self.ds.get_hostname())
